src/main.rs

```diff
@@ -2,33 +2,164 @@
 
 #[macro_use]
 extern crate log;
-
-mod callgraph_matching;
+mod matching;
 mod trawl_source;
 
+use cargo::core::compiler::CompileMode;
 use cargo::core::shell::Shell;
+use cargo::ops::CompileOptions;
+use cargo::{core::resolver::Method, CliError};
+use std::path::PathBuf;
 use structopt::StructOpt;
 
 #[derive(StructOpt, Debug)]
-pub enum Args {
-    Trawl(trawl_source::TrawlArgs),
-    Trace(callgraph_matching::TraceArgs),
+pub struct Args {
+    #[structopt(long = "build_plan")]
+    /// Output a build plan to stdout instead of actually compiling
+    build_plan: bool,
+
+    #[structopt(long = "features", value_name = "FEATURES")]
+    /// Space-separated list of features to activate
+    features: Option<String>,
+
+    #[structopt(long = "all-features")]
+    /// Activate all available features
+    all_features: bool,
+
+    #[structopt(long = "no-default-features")]
+    /// Do not activate the `default` feature
+    no_default_features: bool,
+
+    #[structopt(
+        long = "manifest-path",
+        value_name = "MANIFEST_PATH",
+        parse(from_os_str)
+    )]
+    /// Path to Cargo.toml
+    manifest_path: Option<PathBuf>,
+
+    #[structopt(long = "jobs", short = "j")]
+    /// Number of parallel jobs, defaults to # of CPUs
+    jobs: Option<u32>,
+
+    #[structopt(long = "verbose", short = "v", parse(from_occurrences))]
+    /// Use verbose cargo output (-vv very verbose)
+    verbose: u32,
+
+    #[structopt(long = "quiet", short = "q")]
+    /// Omit cargo output to stdout
+    quiet: bool,
+
+    #[structopt(long = "color", value_name = "WHEN")]
+    /// Cargo output coloring: auto, always, never
+    color: Option<String>,
+
+    #[structopt(long = "frozen")]
+    /// Require Cargo.lock and cache are up to date
+    frozen: bool,
+
+    #[structopt(long = "locked")]
+    /// Require Cargo.lock is up to date
+    locked: bool,
+
+    #[structopt(short = "Z", value_name = "FLAG")]
+    /// Unstable (nightly-only) flags to Cargo
+    unstable_flags: Vec<String>,
+
+    #[structopt(long = "include-tests")]
+    /// Count unsafe usage in tests.
+    include_tests: bool,
+
+    #[structopt(
+        long = "mangled-callgraph-file",
+        value_name = "FILE",
+        parse(from_os_str)
+    )]
+    /// input callgraph file
+    input_callgraph: PathBuf,
+
+    #[structopt(long = "crate-name", value_name = "NAME")]
+    /// crate name
+    crate_name: String,
 }
 
-fn main() {
-    env_logger::init();
+/// Based on code from cargo-bloat. It seems weird that CompileOptions can be
+/// constructed without providing all standard cargo options, TODO: Open an issue
```

Comment:
What do you mean by `all standard cargo options`?

Reply:
great question? that was a comment in the original code that i copypasted along with the code that i moved btwn files.